pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -90,32 +95,38 @@ func (az *Cloud) GetLoadBalancer(clusterName string, service *v1.Service) (statu
 	}, true, nil
 }
 
-func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (string, error) {
+func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (pipName string, pipResourceGroup string, err error) {
 	loadBalancerIP := service.Spec.LoadBalancerIP
 	if len(loadBalancerIP) == 0 {
-		return getPublicIPName(clusterName, service), nil
+		return getPublicIPName(clusterName, service), "", nil
+	}
+
+	// Override the cluster resource group if public IP resource ID presents
+	resourceGroup := az.ResourceGroup
+	if publicIPAddressResourceGroup(service) != "" {
+		resourceGroup = publicIPAddressResourceGroup(service)
```

Comment:
It's harmless to evaluate twice, but also unnecessary.  Can you do something like `if pipResourceGroup := publicIPAddressResourceGroup(service); pipResourceGroup != "" { resourceGroup = pipResourceGroup }`?

Another possibility could be to make `publicIPAddressResourceGroup` handle the defaulting to `az.ResourceGroup` itself.